bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/ClangdFallbackManager.java

```diff
@@ -60,14 +59,12 @@ class FallbackFlags {
 	@Reference
 	private ICBuildConfigurationManager build;
 	@Reference
-	private InitialUri uri;
-	@Reference
 	private IWorkspace workspace;
 
 	@Override
 	public FallbackFlags getFallbackFlagsFromInitialUri(URI root) {
 		if (isWindows) {
-			return uri.find(root)//
+			return Optional.ofNullable(root)//
 					.flatMap(new UriResource(workspace))//
```

Comment:
This won't work since the rootURI form LSP4E is the URI of the initial project. We need the URI of the  initial file.

We can apply the changes of this PR when LSP4E has accepted my PR [#692](https://github.com/eclipse/lsp4e/pull/692)

Reply:
> We need the URI of the initial file.

Why IProject is not sufficient for us?

Reply:
As I can see it, what we do next is get `IProject` from `IResource` (or `IFile`), isn't it?